extensions/replication/tasks/MultipleBackendTask.js

```diff
@@ -332,6 +332,44 @@ class MultipleBackendTask extends ReplicateObject {
         });
     }
 
+    /**
+     * Get a byte range size for an object of the given content length, such
+     * that the range count does not exceed 1024 elements (i.e.
+     * MPU_GCP_MAX_PARTS).
+     * @param {Number} contentLen - The content length of the whole object
+     * @return {Number} The range size to use
+     */
+    _getGCPRangeSize(contentLen) {
+        const pow = Math.pow(2, Math.ceil(Math.log(contentLen) / Math.log(2)));
+        const rangeSize = Math.ceil(pow / MPU_GCP_MAX_PARTS);
+        return rangeSize;
+    }
+
+    /**
+     * Get a byte range size for an object of the given content length, such
+     * that the range size sums to the content length when multiplied by a value
+     * between 2 and 1000, excepting a very large object. This has the effect
+     * of creating MPU parts of the given range size.
+     * @param {Number} contentLen - The content length of the whole object
+     * @return {Number} The range size to use
+     */
+    _getRangeSize(contentLen) {
+        let rangeSize = (1024 * 1024) * 16; // Default 16MB part size.
+        if (contentLen <= rangeSize) {
+            rangeSize = (1024 * 1024) * 5; // AWS minimum MPU part size.
```

Comment:
I don't think you need that special case, if one wants to create an MPU of 10MB he could get that MPU with just one part of 10MB? If you think it's important to have multiple parts, then you would also need to take care of object sizes between 16MB to 32 or 48MB as well to give them a "reasonable" number of parts, but I don't think it's necessary.

Reply:
I'll have to double check this, but I think you need at least two parts to complete an MPU.

Reply:
In that case, ok for me

Reply:
I didn't realize this: it is possible to have an MPU of 1 part, and being the last part it can be < 5MB. I've updated to just return the content length as the range size if <= to 16MB.